### DISCUSSION TITLE
HTTP Sender Retries

### DISCUSSION THREAD

**Discussion by JJWest12** [nextgenhealthcare/connect#5885] [Q&A] [answered]
Is there any way to make an HTTP Sender destination work in a way where it will only execute the retries if a specific error comes up? For example, I am working with an API call that will sometimes return a "No Data" "error" which isn't really an error if no data exists, so the API will retry X amount of times. Is there a way to avoid this?

**Comment by pacmano1** [comment, marked as answer]
Yes, set the response to RAW and handle setting response to "SENT" "ERROR" "QUEUED" as you see fit.   

You need to handle everything in that case including the response code (e.g. the 200, the 401), what to set the response status to, JSON.parse  the message and so on.

**Reply by JJWest12** [reply]
I was unaware that the response transformer would run after each retry, even if the call errors. If that is indeed the case then I should be all set

**Reply by pacmano1** [reply]
It will always if it is set to RAW.  It will not if you leave it as is.

**Reply by pacmano1** [reply]
@man5tg don't hijack threads, create a new one.
